rcl/src/rcl/subscription.c

```diff
@@ -630,7 +630,8 @@ rcl_take_loaned_message(
   // Call rmw_take_with_info.
   bool taken = false;
   rmw_ret_t ret = rmw_take_loaned_message_with_info(
-    subscription->impl->rmw_handle, loaned_message, &taken, message_info_local, allocation);
+    subscription->impl->rmw_handle, (void **) loaned_message, &taken, message_info_local,
```

Comment:
It seems to me that rather than doing this cast here, we should first change `rmw_take_loaned_message_with_info` to take a `const void ** loaned_message`.  At that point, everything will be const and no casts will be needed.

Same goes below for `rmw_return_loaned_message_from_subscription`.